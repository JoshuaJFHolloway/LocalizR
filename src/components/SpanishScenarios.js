import React from 'react'
import SubmitButton from './SubmitButton'

const SpanishScenarios = props => {
  return (
    <SubmitButton
      buttonName = {"Spanish Scenario 1"}
      handleDataSubmitted = {props.handleScenarioClicked}
    />
  )
};

export default SpanishScenarios;