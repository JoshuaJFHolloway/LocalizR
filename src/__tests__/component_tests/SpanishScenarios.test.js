import React from 'react';
import { shallow } from 'enzyme';
import SpanishScenarios from '../../components/SpanishScenarios';


describe('SpanishScenarios', () => {

  let spanishscenario = shallow(<SpanishScenarios />)
  
  describe ('renders components', () => {
    it('renders correctly', () => {
      expect(spanishscenario).toMatchSnapshot();  
    });

    it('renders two SubmitButton component', () => {
      expect(spanishscenario.find('SubmitButton').length).toEqual(2);
    });

  });
});