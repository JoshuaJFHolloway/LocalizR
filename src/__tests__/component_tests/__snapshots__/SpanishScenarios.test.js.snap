// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanishScenarios renders components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpanishScenarios />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SubmitButton
          buttonName="Spanish Scenario 1"
          handleDataSubmitted={undefined}
/>,
        <SubmitButton
          buttonName="Spanish Scenario 2"
          handleDataSubmitted={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonName": "Spanish Scenario 1",
          "handleDataSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonName": "Spanish Scenario 2",
          "handleDataSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SubmitButton
            buttonName="Spanish Scenario 1"
            handleDataSubmitted={undefined}
/>,
          <SubmitButton
            buttonName="Spanish Scenario 2"
            handleDataSubmitted={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonName": "Spanish Scenario 1",
            "handleDataSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonName": "Spanish Scenario 2",
            "handleDataSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
